ethportal-peertest/src/scenarios/find.rs

```diff
@@ -107,3 +109,39 @@ pub fn test_trace_recursive_find_content_local_db(
     assert_eq!(result["content"], json!(HISTORY_CONTENT_VALUE.to_string()));
     assert_eq!(result["route"], json!([]));
 }
+
+pub async fn test_recursive_find_nodes_self(_peertest_config: PeertestConfig, peertest: &Peertest) {
+    let target_node_id = hex_encode(peertest.bootnode.enr.node_id().raw());
+    let request = JsonRpcRequest {
+        method: "portal_historyRecursiveFindNodes".to_string(),
+        id: 15,
+        params: Params::Array(vec![json!(target_node_id)]),
+    };
+    let result = make_ipc_request(&peertest.bootnode.web3_ipc_path, &request).unwrap();
```

Comment:
```suggestion
   let client = reth_ipc::client::IpcClientBuilder::default()
        .build(&peertest.bootnode.web3_ipc_path)
        .await
        .unwrap();
  let result: Vec<Enr> = client.recursive_find_nodes(node_id).await.unwrap();
```
I'm partial to this syntax over the `JsonRpcRequest`/`make_ipc_request` syntax because it looks nicer and the deserialization from json is tested as well.

Reply:
Sure, might as well create the issue now. We don't really need agreement about whether we're going to do it, it's just good to have a place to discuss the refactor. It's even valuable to have an issue where we decide not to do something and simply close it, for our future reference.

Reply:
I went ahead and implemented the refactor here #666